### PR TITLE
Add docformatter to format plain text in docstrings

### LINKED ISSUE #497: Add blackdoc and docformatter to wrap docstrings to 79 chars
**Description of the desired feature**

[`blackdoc`](https://github.com/keewis/blackdoc) is like [`black`](https://github.com/psf/black), but for docstrings! This will help with applying #384, and make our docstrings consistent with our code. 

Would prefer for someone new-ish to take this on. Some knowledge of git commit-hooks would be useful, but you can see https://github.com/pydata/xarray/pull/4177 as an example on how another library has implemented it.

**Are you willing to help implement and maintain this feature?** No

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Add [`docformatter`](https://github.com/myint/docformatter) to format plaintext in docstrings, as mentioned in #497.

docformatter follows PEP 257, which is compatible with numpydoc style. 

EDIT: There may be some incompatibility. The new documentation will look messy if `--pre-summary-newline` is not used.

**Preview the documentation:** https://pygmt-git-docstring-formatter.gmt.vercel.app/

Fixes #497.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: Makefile
Comment (seisman) [inline Makefile:11]: The default lengths are 79 chars for summary and 72 chars for description. Here I choose 79 for both. There are also some options to add more blank lines before/after summary/description. Not sure if we need them.

Comment (weiji14) [reply Makefile:11]:
```suggestion
DOCFORMATTER_OPTIONS=--recursive --pre-summary-newline --wrap-summaries 79 --wrap-descriptions 79
```
Add this to reduce the diff size (Maintains the newline after the triple quotes), and also looks better IMO.

> The default lengths are 79 chars for summary and 72 chars for description. Here I choose 79 for both.

Sounds good. Took me a while to understand the difference between a summary and a description.

> There are also some options to add more blank lines before/after summary/description. Not sure if we need them.

Can probably skip this, prefer to keep things fairly short.

Comment (seisman) [reply Makefile]: All are good points!

File: pygmt/_version.py
Comment (seisman) [inline pygmt/_version.py:111]: The `_version.py` file was generated by [`versioneer`](https://github.com/python-versioneer/python-versioneer), and we should exclude this file. However, the latest docformatter v1.3.1 doesn't provide the `--exclude` option, although its master version does.

Comment (weiji14) [reply pygmt/_version.py:111]: Ideally we would switch from versioneer to setuptools-scm (see https://github.com/pydata/xarray/issues/2853), but haven't got around to doing this yet :slightly_smiling_face:

File: pygmt/tests/test_grdcut.py
Comment (weiji14) [inline pygmt/tests/test_grdcut.py:20]:
```suggestion
    """grdcut an input grid file, and output to a grid file."""
```
Small typo

File: pygmt/_version.py
Comment (seisman) [reply pygmt/_version.py:111]: Yes, I also don't like versioneer, which adds too many files to the project and is less maintained until recently.

File: pygmt/tests/test_grdcut.py
Comment (seisman) [reply pygmt/tests/test_grdcut.py]: Nice catch.

File: pygmt/_version.py
Comment (weiji14) [reply pygmt/_version.py:111]: Let's open up a separate issue/PR to tackle the versioneer->setuptools-scm migration.

Comment (seisman) [reply pygmt/_version.py:111]: OK. Then this PR is ready for review.

**Review by weiji14 — COMMENTED** [review]
Mixed feelings about using `docformatter` here. There are some aspects of PEP 257 that are nice, while others seem strange (though it might just require some getting used to). As you mentioned too at https://github.com/GenericMappingTools/pygmt/pull/642#discussion_r499293934, the version of docformatter (1.3.1) from PyPI is somewhat out of date (11 Nov 2019) and there's features in the master branch which would be nice to have!

Also a bit wary about adding yet another lint tool into our toolbox since it might put off new contributors. The intention is good (making it easier to style code), but we'll need to remind ourselves to put code before formatting (as mentioned in https://github.com/GenericMappingTools/pygmt/blob/master/MAINTENANCE.md#reviewing-and-merging-pull-requests).

**Comment by seisman** [comment]
> As you mentioned too at [#642 (comment)](https://github.com/GenericMappingTools/pygmt/pull/642#discussion_r499293934), the version of docformatter (1.3.1) from PyPI is somewhat out of date (11 Nov 2019) and there's features in the master branch which would be nice to have!

Perhaps we can ask if they could have a new release.

> Also a bit wary about adding yet another lint tool into our toolbox since it might put off new contributors.

It seems straightforward to support slash commands like `/format` using [slash-command-dispatch](https://github.com/peter-evans/slash-command-dispatch) (see a demo at https://github.com/peter-evans/slash-command-dispatch/pull/28). Then contributors and even core developer won't need to run `make format` locally.

File: pygmt/clib/session.py
Comment (seisman) [reply pygmt/clib/session.py:341]: I think "summary" can be only one sentence, so the "summary" and "description" is separated at the first "period".

**Comment by weiji14** [comment]
> Perhaps we can ask if they could have a new release.

It's been asked at https://github.com/myint/docformatter/issues/55.

> It seems straightforward to support slash commands like `/format` using [slash-command-dispatch](https://github.com/peter-evans/slash-command-dispatch) (see a demo at [peter-evans/slash-command-dispatch#28](https://github.com/peter-evans/slash-command-dispatch/pull/28)). Then contributors and even core developer won't need to run `make format` locally.

Ah that does look neat!

**Comment by seisman** [comment]
> > Perhaps we can ask if they could have a new release.
> 
> It's been asked at [myint/docformatter#55](https://github.com/myint/docformatter/issues/55).

Let's keep this PR open for a while.

File: pygmt/_version.py
Comment (seisman) [reply pygmt/_version.py]: We switched from versioneer to setuptools-scm in PR #695, so we don't need to wait for the `--exclude` feature in docformatter.

File: pygmt/sphinx_gallery.py
Comment (weiji14) [reply pygmt/sphinx_gallery.py:22]: Once https://github.com/myint/docformatter/pull/49 is merged (and a `docformatter` release is made), then we can rerender things and hopefully get a smaller diff.

Comment (weiji14) [reply pygmt/sphinx_gallery.py]: Docformatter 1.4 has been released at https://pypi.org/project/docformatter/1.4/ so we can try to finish up this PR.

File: Makefile
Comment (weiji14) [inline Makefile:10]:
```suggestion
DOCFORMATTER_OPTIONS=--recursive --pre-summary-newline --make-summary-multi-line --wrap-summaries 79 --wrap-descriptions 79
```
Do we want to force `--make-summary-multi-line`? This will turn every description (even short ones) into:
```python
"""
This is a description.
"""
```

Comment (seisman) [reply Makefile:10]: Yes, it looks cleaner.

Comment (seisman) [reply Makefile]: Just find that [PEP 257](https://www.python.org/dev/peps/pep-0257/) says:

> Unless the entire docstring fits on a line, place the closing quotes on a line by themselves.

Comment (weiji14) [reply Makefile]: Hmm, seems like `docformatter` didn't implement that [one-line docstring PEP 257 guideline](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings) correctly? I think it's ok to just force multi-line though, as long as there's an automated solution.

Comment (weiji14) [reply Makefile]: /format

File: Makefile
Comment (weiji14) [inline Makefile:20]:
```suggestion
	@echo "  check     run code style and quality checks (black, blackdoc, docformatter, flake8 and isort)"
```

File: environment.yml
Comment (weiji14) [inline environment.yml:24]: This environment.yml file should be alphabetically ordered.

File: requirements-dev.txt
Comment (weiji14) [inline requirements-dev.txt:12]: Ensure that this file is alphabetically ordered too.

**Review by weiji14 — COMMENTED** [review]
Just one nitpick on alphabetically ordering, otherwise should be good to merge once the Style Checks tests pass.

**Comment by weiji14** [comment]
> /format

Might need to do it manually locally for now, the Github Action is using the workflow yaml file from the master branch.

File: pygmt/tests/test_figure.py
Comment (weiji14) [inline pygmt/tests/test_figure.py:3]:
```suggestion
Tests the behavior of the Figure class.

Doesn't include the plotting commands which have their own test files.
```
One last typo to fix :slightly_smiling_face:

**Review by weiji14 — APPROVED** [review]

**Pull request merged**
